plugins/modules/dcnm_image_upload.py

```diff
@@ -0,0 +1,777 @@
+
+        if self.image_upload_info == []:
+            # No filters specified. Add all images to the output
+            self.result["response"].extend(dcnm_image_list)
+
+        for elem in self.image_upload_info:
+            # Image names a re provided as filters. Filter the output as required
+            match_elem = [
+                img
+                for img in dcnm_image_list
+                if ((os.path.basename(elem["name"]) == img["imageName"]))
+            ]
+            if match_elem:
+                self.result["response"].append(match_elem[0])
+
+    def dcnm_image_upload_get_want(self):
+
+        """
+        This routine updates self.want with the payload information based on the playbook configuration.
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+
+        if [] is self.files:
+            return
+
+        if not self.image_upload_info:
+            return
+
+        for elem in self.image_upload_info:
+
+            payload = self.dcnm_image_upload_get_payload(elem)
+            if payload not in self.want:
+                self.want.append(payload)
+
+    def dcnm_image_upload_get_have(self):
+
+        """
+        Routine to get exisitng image_upload information from DCNM that matches information in self.want.
+        This routine updates self.have with all the image_upload that match the given playbook configuration
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+
+        if self.want == []:
+            return
+
+        dcnm_image_list = self.dcnm_image_upload_get_info_from_dcnm()
+
+        # Compare the images from want and dcnm_image_list. Keep only those that match
+        for want in self.want:
+            match_have = [
+                elem
+                for elem in dcnm_image_list
+                if os.path.basename(want["filePath"]) == elem["imageName"]
+            ]
+
+            if match_have and match_have[0]["imageName"] not in self.have:
+                self.have.append(match_have[0])
+
+    def dcnm_image_upload_validate_query_state_input(self, cfg):
+
+        """
+        Playbook input will be different for differnt states. This routine validates the query state
+        input. This routine updates self.image_upload_info with validated playbook information related
+        to query state.
+
+        Parameters:
+            cfg (dict): The config from playbook
+
+        Returns:
+           None
+        """
+
+        arg_spec = dict(name=dict(required=True, type="str"))
+
+        image_upload_info, invalid_params = validate_list_of_dicts(
+            cfg, arg_spec
+        )
+        if invalid_params:
+            mesg = "Invalid parameters in playbook: {0}".format(invalid_params)
+            self.module.fail_json(msg=mesg)
+
+        if image_upload_info:
+            self.image_upload_info.extend(image_upload_info)
+
+    def dcnm_image_upload_validate_input(self, cfg):
```

Comment:
Maybe add a docstring later.  Would be good to include in docstrings if a method calls fail_json() or if it raises an exception (in which case, the type of exception(s) raised is useful to have in the docstring, since it gives callers a heads-up as to what can fail when calling a given method).